src/coreclr/src/vm/gccover.cpp

```diff
@@ -744,23 +744,37 @@ void replaceSafePointInstructionWithGcStressInstr(UINT32 safePointOffset, LPVOID
     //Determine if instruction before the safe point is call using immediate (BLX Imm)  or call by register (BLX Rm)
     BOOL  instructionIsACallThroughRegister = FALSE;
     BOOL instructionIsACallThroughImmediate = FALSE;
+
 #if defined(TARGET_ARM)
 
+    // POSSIBLE BUG: Note that we are looking backwards by 2 or 4 bytes, looking for particular call instruction encodings.
+    // However, we don't know if the previous instruction is 2 bytes or 4 bytes. Looking back 2 bytes could be looking into
+    // the middle of a 4-byte instruction. The only safe way to do this is by walking forward from the first instruction of
+    // the function.
+
     // call by register instruction is two bytes (BL<c> Reg T1 encoding)
     WORD instr = *((WORD*)savedInstrPtr - 1);
-
     instr = instr & 0xff87;
-    if((instr ^ 0x4780) == 0)
+    if ((instr ^ 0x4780) == 0)
+    {
         // It is call by register
         instructionIsACallThroughRegister = TRUE;
+    }
+    else
```

Comment:
The only code change here is putting the `instructionIsACallThroughImmediate` case as an `else` clause, if we didn't first see a `blx reg` instruction. The rest is just whitespace and semicolons.